lib/npmdc/checkers/yarn/yarn_checker.rb

```diff
@@ -0,0 +1,144 @@
+require 'pty'
+require 'npmdc/checkers/yarn/yarn_installation_checker'
+
+module Npmdc
+  class YarnChecker
```

Comment:
Maybe we can extract the common checker functionality into a base class and inherit from it? And also put checkers into a namespace, say:

```ruby
# npmdc/checkers/base.rb
module Npmdc
  module Checkers
    class Base
       extend Forwardable
       include Npmdc::Errors
       
       delegate ...
    end
  end
end

# npmdc/checkers/yarn.rb
module Npmdc
  module Checkers
    class Yarn < Base
    end
  end
end
```

/cc @lysyi3m  @sponomarev